repository/src/main/java/org/apache/atlas/discovery/EntityLineageService.java

```diff
@@ -220,24 +220,60 @@ public AtlasLineageListInfo getLineageListInfoOnDemand(String guid, LineageListR
         return ret;
     }
 
-    private HashMap<String, Boolean> validateAndGetEntityTypeMap(String guid) throws AtlasBaseException {
+    public class EntityValidationResult {
+        public final boolean isProcess;
+        public final boolean isDataSet;
+        public final boolean isConnection;
+        public final boolean isConnectionProcess;
+        public final boolean isDataProduct;
+
+        public EntityValidationResult(boolean isProcess, boolean isDataSet, boolean isConnection, boolean isConnectionProcess, boolean isDataProduct) {
+            this.isProcess = isProcess;
+            this.isDataSet = isDataSet;
+            this.isConnection = isConnection;
+            this.isConnectionProcess = isConnectionProcess;
+            this.isDataProduct = isDataProduct;
+        }
+
+        public boolean CheckIfLineageRequestIsForAssetTypeProcess(String lineageType){
```

Comment:
Please refactor the name, it seems confusing a bit. Would recommend 'CheckIfConnectorVertex' as it is being used to check if the entity being handled is a connector or data vertex. Correct me if you think otherwise.